dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java

```diff
@@ -0,0 +1,301 @@
+                        nav = new NavResultHydrated(nav, this.context);
+                        children.add(nav);
+                    }
+                } else if (item instanceof Link) {
+                    Link itemLink = (Link) item;
+                    NavResult nav = new NavResult(folder.getInode(), host.getIdentifier(), languageId);
+
+                    if (itemLink.getLinkType()
+                        .equals(LinkType.CODE.toString()) && LinkType.CODE.toString() != null) {
+                        nav.setCodeLink(itemLink.getLinkCode());
+                    } else {
+                        nav.setHref(itemLink.getWorkingURL());
+                    }
+                    nav.setTitle(itemLink.getTitle());
+                    nav.setOrder(itemLink.getSortOrder());
+                    nav.setType("link");
+                    nav.setTarget(itemLink.getTarget());
+                    nav.setPermissionId(itemLink.getPermissionId());
+                    nav.setShowOnMenu(itemLink.isShowOnMenu());
+                    nav = new NavResultHydrated(nav, this.context);
+                    children.add(nav);
+                } else if (item instanceof IFileAsset) {
+                    IFileAsset itemFile = (IFileAsset) item;
+
+                    if (itemFile.getLanguageId() == languageId || LanguageWebAPI.canDefaultFileToDefaultLanguage()) {
+                        ident = APILocator.getIdentifierAPI()
+                            .find(itemFile.getPermissionId());
+                        NavResult nav = new NavResult(folder.getInode(), host.getIdentifier(), languageId);
+
+                        nav.setTitle(itemFile.getFriendlyName());
+                        nav.setHref(ident.getURI());
+                        nav.setOrder(itemFile.getMenuOrder());
+                        nav.setType("file");
+                        nav.setPermissionId(itemFile.getPermissionId());
+                        nav.setShowOnMenu(itemFile.isShowOnMenu());
+                        nav = new NavResultHydrated(nav, this.context);
+                        children.add(nav);
+                    }
+                }
+            }
+
+            CacheLocator.getNavToolCache()
+                .putNav(host.getIdentifier(), folder.getInode(), result, languageId);
+
+            return new NavResultHydrated(result, this.context);
+        }
+    }
+
+    /**
+     * Pass the level of the nav you wish to retrieve, based on the current path, level 0 being the
+     * root
+     * 
+     * @param level
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public NavResult getNav(int level) throws DotDataException, DotSecurityException {
+        if (level < 1)
+            return getNav("/");
+
+
+        String reqPath = getNav().getHref();
+
+        String[] levels = reqPath.split("/");
+
+
+        if (level + 1 > levels.length)
+            return null;
+
+        StringBuffer sw = new StringBuffer();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "StringBuffer" by an unsynchronized one such as "StringBuilder". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)